hydromt/models/model.py

```diff
@@ -470,7 +388,6 @@ def write_data_catalog(
             cat.add_source(name, source)
         # write data catalog
         if cat.sources:
-            self._assert_write_mode()
```

Comment:
did you replace that check somewhere else then? Using self.root.is_writting_mode()

Reply:
Yes, this moved to `ModelComponent._assert_writing_mode()`.